### PR TITLE
[WIP] add support for Azure provisioning with Rigger

### PR DESCRIPTION
**Pull request by sgoings**

### CODE REVIEW

File: contrib/azure/create-azure-user-data [hunk 1]
```diff
@@ -56,14 +56,19 @@ def main():
     azure_template = get_file("azure-user-data-template")
     coreos_template = get_file("../coreos/user-data.example")
 
-    configuration_coreos_template = yaml.safe_load(coreos_template)
+    coreos_template_w_url = None
+    with open(coreos_template.name, 'r') as file :
+      coreos_template_w_url = file.read()
+
+    coreos_template_w_url = coreos_template_w_url.replace('#DISCOVERY_URL', url)
```
Comment (sgoings) [inline contrib/azure/create-azure-user-data]: With newer versions of PyYAML the "safe_load" would set the `discovery` variable in cloud-config to "none" and remove the comment, making this logic ineffective. This was my workaround.

**Comment by carmstrong** [comment]
I don't know much about the Standard_DS2 instance type - is that the new recommended default? Also, I don't want to change the heartbeat timeouts. We don't customize that on any other platform, and I'd really like to avoid that. 

I'd like to see this run with premium storage and see how it does.

**Comment by sgoings** [comment]
@carmstrong I'm using the Standard_DS2 instance type as per our conversation with the Azure folks.

I'm currently running tests every hour on https://ci.deis.io/job/rigger-provision-azure... I'd like to see a run pass nicely and then I'll start to back out changes in the following order:

- heartbeat timeouts
- instance type change

**Comment by carmstrong** [comment]
@sgoings Sounds like a plan. :+1:

**Comment by carmstrong** [comment]
Why are we renaming `arm-template.json` to `azuredeploy.json`? I'd also still like to drop the customized timeouts.

**Comment by sgoings** [comment]
It's still a WIP. :smile:

**Comment by carmstrong** [comment]
It's a passing WIP ;)

**Comment by sgoings** [comment]
MERGE IT! jk jk.

**Comment by mboersma** [comment]
Can this include support for Azure availability sets as suggested in #4322?

**Comment by mboersma** [comment]
@sgoings do you think this could be brought over the finish line in the next week or two so it can go in the Deis v1 LTS release? I know you're busy with Deis v2 tasks, so if this isn't on your near-term roadmap we may want to close it.

**Comment by sgoings** [comment]
Yeah we could probably get it into the LTS release.

**Comment by bacongobbler** [comment]
ping @sgoings, any progress on this or is it non-critical for LTS? :)

**Comment by sgoings** [comment]
This is definitely non critical for LTS.

**Comment by mboersma** [comment]
Azure testing efforts are now targeted at Deis v2. It's unlikely we'll revive this PR so I'm closing it.

**Pull request closed without merging**